File: src/main/java/io/koju/autopos/config/AuditingConfiguration.java
package io.koju.autopos.config;

import io.koju.autopos.security.SecurityUtils;
import io.koju.autopos.user.domain.User;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.domain.AuditorAware;
import org.springframework.data.jpa.repository.config.EnableJpaAuditing;

@Configuration
@EnableJpaAuditing
public class AuditingConfiguration {

    @Bean
    public AuditorAware<User> auditorAware() {
        return SecurityUtils::getCurrentUser;
    }
}

File: src/main/scala/io/koju/autopos/config/AuditingConfiguration.scala
package io.koju.autopos.config

import io.koju.autopos.security.SecurityUtils
import io.koju.autopos.user.domain.User
import org.springframework.context.annotation.{Bean, Configuration}
import org.springframework.data.domain.AuditorAware
import org.springframework.data.jpa.repository.config.EnableJpaAuditing

@Configuration
@EnableJpaAuditing
class AuditingConfiguration {

  @Bean
  def auditorAware: AuditorAware[User] = new AuditorAware[User] {
    override def getCurrentAuditor = SecurityUtils.getCurrentUser
  }
}